core/mondoo-azure-security.mql.yaml

```diff
@@ -491,17 +484,16 @@ queries:
     impact: 80
     mql: |
       azure.subscription.network.securityGroups {
```

Comment:
Suggest using the `.all()` statement as with the other PR:
```
testNSG = azure.subscription.network.securityGroup.securityRules
        .where(
          properties.access == 'Allow'
            && direction == 'Inbound'
            && properties.protocol == /TCP|\*/
            && properties.sourceAddressPrefix == /\*|0\.0\.0\.0|<nw>\/0|\/0|internet|any/
        )

      testNSG.all(properties.destinationPortRange != "*")

      a = testNSG.all(destinationPortRange.all(fromPort < 3389 && toPort < 3389))

      b = testNSG.all(destinationPortRange.all(fromPort > 3389 && toPort > 3389))

      isPortRangeValid = a || b
      isPortRangeValid
      
```

Reply:
Thank you for the suggestion. However, it appears that it doesn't effectively address all the test scenarios we have. I appreciate your input and look forward to exploring alternative solutions or Keeping the Original one that might be more comprehensive.